dev-util/clion/clion-2025.1-r1.ebuild

```diff
@@ -0,0 +1,150 @@
+	my_arch_radler=$(usex amd64 x64 arm64)
+	local other_arch_amd64_aarch64=$(usex amd64 aarch64 amd64)
+	local other_arch_radler=$(usex amd64 arm64 x64)
+	tc-export OBJCOPY
+	default
+
+	local remove_me=(
+		Install-Linux-tar.txt
+		help/ReferenceCardForMac.pdf
+		bin/cmake
+		bin/gdb/linux
+		bin/lldb/linux
+		bin/ninja
+		plugins/remote-dev-server/selfcontained
+	)
+	remove_me+=(
+		lib/async-profiler/${other_arch_amd64_aarch64}
+		plugins/clion-radler/DotFiles/linux-${other_arch_radler}
+		plugins/clion-radler/dotTrace.dotMemory/DotFiles/linux-${other_arch_radler}
+		plugins/python-ce/helpers/pydev/pydevd_attach_to_process/attach_linux_${other_arch_amd64_aarch64}.so
+	)
+	use !amd64 && remove_me+=(
+		plugins/python-ce/helpers/coveragepy_old/coverage/tracer.cpython-310-x86_64-linux-gnu.so
+		plugins/python-ce/helpers/pydev/pydevd_attach_to_process/attach_linux_x86.so
+	)
+
+	rm -rv "${remove_me[@]}" || die
+
+	# excepting files that should be kept for remote plugins
+	skip_nonx86_files=(
+		"plugins/platform-ijent-impl/ijent-$(usex amd64 aarch64 x86_64)-unknown-linux-musl-release"
+		"plugins/clion-radler/DotFiles/linux-musl-${other_arch_radler}/jb_zip_unarchiver"
+		"plugins/clion-radler/DotFiles/linux-arm/jb_zip_unarchiver"
+		"plugins/clion-radler/DotFiles/linux-musl-arm/jb_zip_unarchiver"
+		"plugins/gateway-plugin/lib/remote-dev-workers/remote-dev-worker-linux-$(usex amd64 arm64 amd64)"
+	)
+	# removing debug symbols and relocating debug files as per #876295
+	# we're escaping all the files that contain $() in their name
+	# as they should not be executed
+	find . -type f ! -name '*$(*)*' -print0 | while IFS= read -r -d '' file; do
+		for skip in "${skip_nonx86_files[@]}"; do
+			[[ ${file} == "./${skip}" ]] && continue 2
+		done
+		if file "${file}" | grep -qE "ELF (32|64)-bit"; then
+			${OBJCOPY} --remove-section .note.gnu.build-id "${file}" || die
+			debugedit -b "${EPREFIX}/opt/${PN}" -d "/usr/lib/debug" -i "${file}" || die
+		fi
+	done
+
+	patchelf --set-rpath '$ORIGIN' "jbr/lib/libjcef.so" || die
+	patchelf --set-rpath '$ORIGIN' "jbr/lib/jcef_helper" || die
+	patchelf --set-rpath '$ORIGIN/../lib' "bin/clang/linux/${my_arch_suffix}/lib/libclazyPlugin.so" || die
+}
+
+src_install() {
+	local dir="/opt/${PN}"
+
+	insinto "${dir}"
+	doins -r *
+	fperms 755 "${dir}"/bin/{clion.sh,format.sh,fsnotifier,inspect.sh,jetbrains_client.sh,ltedit.sh,remote-dev-server.sh,restarter,clang/linux/${my_arch_suffix}/bin/{clangd,clang-tidy,clazy-standalone,llvm-symbolizer}}
+
+	if [[ -d jbr ]]; then
+		fperms 755 "${dir}"/jbr/bin/{java,javac,javadoc,jcmd,jdb,jfr,jhsdb,jinfo,jmap,jps,jrunscript,jstack,jstat,keytool,rmiregistry,serialver}
+		# Fix #763582
+		fperms 755 "${dir}"/jbr/lib/{chrome-sandbox,jcef_helper,jexec,jspawnhelper}
+	fi
+
+	fperms 755 "${dir}"/plugins/clion-radler/DotFiles/linux-${my_arch_radler}/Rider.Backend
+
+	dosym -r "${EPREFIX}/usr/bin/ninja" "${dir}"/bin/ninja/linux/${my_arch_suffix}/ninja
```

Comment:
I think `dosym -r` does not need `${EPREFIX}` - especially since you didn't use it for both arguments

Reply:
Also not a part of arm64 diff

Reply:
> I think `dosym -r` does not need `${EPREFIX}` - especially since you didn't use it for both arguments

I think this is here for a very, very long time :smile: 

@WhatAmISupposedToPutHere you can change this, if you wish.